src/kohlrahbi/ahb/__init__.py

```diff
@@ -189,7 +189,9 @@ def extract_pruefis_from_table(table: Table) -> list[str]:
 
 def table_header_contains_text_pruefidentifikator(table: Table) -> bool:
     """Checks if the table header contains the text 'Prüfidentifikator'."""
-    return table.row_cells(0)[-1].paragraphs[-1].text.startswith("Prüfidentifikator")  # type:ignore[no-any-return]
+    pattern = r"Prüfidentifikator(?:\t){0,10}\t\d+"
+    # "matches "Prüfidentifikator" followed by at least 1 tab separated numbers, max 11 pruefis is chosen arbitrarily
+    return bool(re.search(pattern, table.row_cells(0)[-1].text))
```

Comment:
@hf-kklein gab es hier nicht noch ein anderes pattern was das pattern irgendwie compiled hat? 
Ziel war es dass das regex nur einmal ausgeführt werden muss. 

Oft haben wir das pattern global ausgelagert und dann in der Funktion nur angewendet. 

Weißt du was ich meine?

Reply:
my_pattern = re.comile(r"Prüfidentifikator(?:\t){0,10}\t\d+")

und dann mypattern.search(...)

Reply:
bringt dann was, wenn die funktion sehr oft aufgerufen wird. wobei auch so der regex-teil einen internen cache hat iirc.

Reply:
Ah okay. Die Funktion wir in jeder Tabelle aufgerufen. Aber wenn sie einen internen Cache hat passt das.